backend/baseclass.pm

```diff
@@ -362,6 +365,33 @@ sub start_encoder {
     return;
 }
 
+sub _stop_video_encoder {
+    my ($self) = @_;
+
+    my $video_encoder_pids = $self->{video_encoder_pids};
+    return undef unless keys %$video_encoder_pids;
+
+    # give the video encoder processes 30 seconds to finalize the video
+    # note: Sending SIGTERM only once because successive SIGTERM signals might be interpreted by the video encoder as
+    #       request to stop it forcefully.
+    no autodie qw(kill waitpid);
+    kill TERM => (keys %$video_encoder_pids);
+    bmwqemu::diag 'Waiting for video encoder to finalize the video';
+    for (my $timeout = 30; $timeout > 0; sleep($timeout -= 1)) {
+        for my $pid (keys %$video_encoder_pids) {
+            my $ret = waitpid($pid, WNOHANG);
```

Comment:
That's now a rather low-level approach of what should be handled by the process management framework. Maybe we need to use an alternative to Mojo::IOLoop::ReadWriteProcess with just Mojo::IOLoop or also IPC::Run how I recently have done in openQA tests replacing manual fork calls

Reply:
I've been fighting against Mojo::IOLoop::ReadWriteProcess long enough so I though let's go back to the roots. Besides: Mojo::IOLoop is not used here in favor of our own `run_capture_loop`. Switching to Mojo::IOLoop in a consistent way would mean rewriting that and I don't think it is worth it. The process handling code within the backend is also generally low-level. We're using the pipes directly (within that `run_capture_loop`) and I don't want any framework to interfere with that.